pkg/provider/provider.go

```diff
@@ -52,13 +46,247 @@ func Provider() *schema.Provider {
 				Sensitive:     true,
 				ConflictsWith: []string{"browser_auth", "private_key_path", "private_key", "private_key_passphrase", "oauth_access_token", "oauth_refresh_token"},
 			},
+			// todo: add database and schema once unqualified identifiers are supported
+			"warehouse": {
+				Type:        schema.TypeString,
+				Description: "Specifies the virtual warehouse to use by default for queries, loading, etc. in the client session. Can be sourced from SNOWFLAKE_WAREHOUSE environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_WAREHOUSE", nil),
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Description: "Specifies the role to use by default for accessing Snowflake objects in the client session. Can be sourced from the `SNOWFLAKE_ROLE` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_ROLE", nil),
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Description: "Snowflake region, such as \"eu-central-1\", with this parameter. However, since this parameter is deprecated, it is best to specify the region as part of the account parameter. For details, see the description of the account parameter. [Snowflake region](https://docs.snowflake.com/en/user-guide/intro-regions.html) to use.  Required if using the [legacy format for the `account` identifier](https://docs.snowflake.com/en/user-guide/admin-account-identifier.html#format-2-legacy-account-locator-in-a-region) in the form of `<cloud_region_id>.<cloud>`. Can be sourced from the `SNOWFLAKE_REGION` environment variable.",
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_REGION", "us-west-2"),
+				Optional:    true,
+				Deprecated:  "Specify the region as part of the account parameter",
```

Comment:
It's deprecated. Shouldn't it be put with other deprecated parameters at the bottom ?

Reply:
done